mbtiles/src/diff.rs

```diff
@@ -0,0 +1,29 @@
+use std::{path::PathBuf, str::FromStr};
+
+use crate::{Mbtiles, MbtError, MbtResult, is_empty_database};
+
+impl Mbtiles {
+    pub async fn diff(&self, dst_path: PathBuf, diff_path: PathBuf) -> MbtResult<()>{
```

Comment:
I think it is better to have a function outside of the MBtiles that takes all parameters, and produces the result.  Mbtiles is like a wrapper for a single file, not all 3. That said, you will need to at some point to open a file, attach A and B to it, and start writing to it.